sdk/eventhub/Azure.Messaging.EventHubs/tests/Producer/EventDataBatchTests.cs

```diff
@@ -22,31 +24,66 @@ public class EventDataBatchTests
         ///   constructor.
         /// </summary>
         ///
+        [Test]
         public void ConstructorVerifiesTheTransportBatch()
         {
-            Assert.That(() => new EventDataBatch(null, new SendEventOptions()), Throws.ArgumentNullException);
+            Assert.That(() => new EventDataBatch(null, "ns", "eh", new SendEventOptions()), Throws.ArgumentNullException);
         }
 
         /// <summary>
         ///   Verifies property accessors for the <see cref="EventDataBatch" />
         ///   constructor.
         /// </summary>
         ///
+        [Test]
+        [TestCase(null)]
+        [TestCase("")]
+        public void ConstructorVerifiesTheFullyQualifiedNamespace(string fullyQualifiedNamespace)
+        {
+            // Seems ExactTypeConstraints is not re-entrant.
+            ExactTypeConstraint TypeConstraint() => fullyQualifiedNamespace is null ? Throws.ArgumentNullException : Throws.ArgumentException;
```

Comment:
We could do this as just `Throws.InstanceOf(Type)`... I know that Heath when this route elsewhere, but I keep meaning to go back and simplify.